scripts/fuzz_shell.js

```diff
@@ -94,21 +113,25 @@ function refreshView() {
 }
 
 // Run the wasm.
-var sortedExports = [];
 for (var e in exports) {
-  sortedExports.push(e);
-}
-sortedExports.sort();
-sortedExports.forEach(function(e) {
-  if (typeof exports[e] !== 'function') return;
+  if (typeof exports[e] !== 'function') {
+    continue;
+  }
+  // Send the function a null for each parameter. Null can be converted without
+  // error to both a number and a reference.
```

Comment:
What does sending a null for a parameter mean? Does this mean the `null` arg in
```js
func.apply(null, args);
```
?

I'm not very familiar with JS still, so I looked it up (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply), and it looks the first param is `thisArg`. Do we need this arg? What does converting this to a number/reference mean?

Reply:
Ah, yeah, this could be confusing because `null` is used in two ways. Yes, the first null is `thisArg`, which we have to pass only because we are using `.apply`. Basically we want to do

```js
func(null, null);
func(null, null, null);
```

for a function with two or three parameters, etc. To do that we want to send a vector of arguments, which means we need to use `apply`, which forces us to provide something for `thisArg`. The value there doesn't matter.

The nulls that we pass in to the params get converted into wasm types. Sending `null` to an `i32` param gives a 0, and to a `(ref null any)` gives `(ref.null any)` and so forth. The specific values don't matter (though it would be nice eventually to send in more values than `0`, but the fuzzer added calls inside the wasm with such things). But we do not want to trap because of a conversion, which would happen if we passed `0` because that can't be converted to a `(ref null any)` for example.

Reply:
I see, thanks!

Reply:
Do we care about supporting non-nullable reference parameters?

Reply:
In calls from JS I'd say no - we don't have a good way to construct those atm. As mentioned above the fuzzer generates internal calls to exports which gives internal coverage for such functions at least (but not for the JS-wasm boundary, which maybe some day will be important here).